### PR TITLE
Some generic impls for Either<Length, T>

### PR DESCRIPTION
**Pull request by ghost**
We seem to be using `parse_non_negative` a lot for `LengthOrFoo` types, so it's better to have generic impls (now, and in the future) for such things (assuming that `Length` will always be in the first variant of `Either`).

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

- [x] These changes do not require tests because it's a refactor

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14296)

### CODE REVIEW

File: components/style/properties/helpers/animated_properties.mako.rs [hunk 1]
```diff
@@ -680,11 +679,11 @@ impl Interpolate for BoxShadow {
     }
 }
 
-impl Interpolate for LengthOrNone {
+impl<T: Copy> Interpolate for Either<Au, T> {
```
Comment (emilio) [inline components/style/properties/helpers/animated_properties.mako.rs]: Why the `T: Copy` bound? Seems quite arbitrary. I'd prefer a generic `Interpolate` for `Either<T, U>`, if that's possible, that may very well not be.

**Review by emilio — COMMENTED** [review]
The `parse_non_negative_length` I guess is fine, I guess, but I'm curious about the reasoning for the other.

File: components/style/properties/helpers/animated_properties.mako.rs (hunk 1 above)
Comment (Wafflespeanut) [inline components/style/properties/helpers/animated_properties.mako.rs]: It doesn't have to be `Copy` (removed it and added a more-or-less generic impl).

File: components/style/properties/helpers/animated_properties.mako.rs [hunk 2]
```diff
@@ -680,11 +679,11 @@ impl Interpolate for BoxShadow {
     }
 }
 
-impl Interpolate for LengthOrNone {
+impl<T: Interpolate, U> Interpolate for Either<T, U> {
```
Comment (emilio) [inline components/style/properties/helpers/animated_properties.mako.rs]: I think this is wrong, what if we have `Either<Length, Number>`? We'd only be interpolating numbers with this code.

**Pull request merged**